BookPlayer/Library/BaseListViewController.swift

```diff
@@ -152,23 +154,57 @@ class BaseListViewController: UIViewController {
         }
     }
 
-    func setupPlayer(books: [Book]) {
-        // Stop setup if no books are passed
+    func queueBooksForPlayback(_ startItem: LibraryItem, forceAutoplay: Bool = false) -> [Book] {
+        var books = [Book]()
+        let isPlaylist = startItem is Playlist
+        let shouldAutoplayLibrary = UserDefaults.standard.bool(forKey: UserDefaultsConstants.autoplayEnabled)
+        let shouldAutoplay = (isPlaylist || shouldAutoplayLibrary || forceAutoplay)
+
+        guard let book = startItem as? Book else {
+            return books
+        }
+
+        books.append(book)
+
+        guard
+            shouldAutoplay,
+            let remainingItems = self.items.split(whereSeparator: { $0 == startItem }).last
+            else {
+                return books
+        }
+
+        for item in remainingItems {
+            if let playlist = item as? Playlist, let playlistBooks = playlist.books?.array as? [Book] {
+                for book in playlistBooks {
```

Comment:
instead of this `for` loop, you can do `books.append(contentsOf: playlistBooks)`

Reply:
Ever learning. Will update.